src/key_io.cpp

```diff
@@ -92,15 +92,15 @@ CTxDestination DecodeDestination(const std::string& str, const CChainParams& par
 
         // Set potential error message.
         // This message may be changed if the address can also be interpreted as a Bech32 address.
-        error_str = "Invalid prefix for Base58-encoded address";
+        error_str = "Invalid prefix for Base58-encoded address. Valid address starts with `1` or `3` (mainnet) and `m` or `n` or `2` (testnet)";
     }
     data.clear();
     const auto dec = bech32::Decode(str);
     if ((dec.encoding == bech32::Encoding::BECH32 || dec.encoding == bech32::Encoding::BECH32M) && dec.data.size() > 0) {
         // Bech32 decoding
         error_str = "";
         if (dec.hrp != params.Bech32HRP()) {
-            error_str = "Invalid prefix for Bech32 address";
+            error_str = "Invalid prefix for Bech32 address. Valid Bech32 address starts with `bc1` (mainnet) and `tb1` (testnet)";
```

Comment:
Why not simply print `dec.hrp != params.Bech32HRP()`?

Reply:
You mean use something like this?

```C++
error_str = "Invalid prefix for Bech32 address. Details:" + dec.hrp + "!=" + params.Bech32HRP();
```

Which will give this error if wrong prefix for a bech32 address is used:

![image](https://user-images.githubusercontent.com/13405205/115749372-a73a7900-a3b4-11eb-9548-6e2ec1bf8a9a.png)

Looks more confusing for some users.

Reply:
Maybe this works better:

```C++
error_str = "Invalid prefix `" + dec.hrp + "`for Bech32 address. Suggested prefix: `"  + params.Bech32HRP() + "`";
```

Reply:
'Suggested prefix' implies you (the user) can just change it, which obviously isn't the case. I think the error message does need to emphasise the fact you might be using an address from the wrong network.

Reply:
I wanted to mention "Wrong network" in this error but I am not sure if wrong network is the only reason why a prefix can be wrong and condition in `if` statement at L102 becomes true.

Reply:
@prayank23 if the decoding succeeded but the HRP is wrong then I think "Wrong network" is a valid assumption. It would either be that or the input was another type of data entirely, using the same checksum, which is much more unlikely and can be safely ignored.